src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerInitializationHostService.cs

```diff
@@ -0,0 +1,118 @@
+{
+    public class LinuxContainerInitializationHostService : IHostedService
+    {
+        private readonly ScriptSettingsManager _settingsManager;
+        private readonly IInstanceManager _instanceManager;
+        private readonly ILogger _logger;
+        private CancellationToken _cancellationToken;
+
+        public LinuxContainerInitializationHostService(ScriptSettingsManager settingsManager, IInstanceManager instanceManager, ILoggerFactory loggerFactory)
+        {
+            _settingsManager = settingsManager;
+            _instanceManager = instanceManager;
+            _logger = loggerFactory.CreateLogger(ScriptConstants.LogCategoryHostGeneral);
+        }
+
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Initializing LinuxContainerInitializationService.");
+            _cancellationToken = cancellationToken;
+            await Run();
+        }
+
+        private async Task Run()
+        {
+            if (_settingsManager.IsLinuxContainerEnvironment)
+            {
+                await InitializeAssignmentContext();
+            }
+        }
+
+        private async Task InitializeAssignmentContext()
+        {
+            var startContext = _settingsManager.GetSetting(EnvironmentSettingNames.ContainerStartContext);
+
+            // Container start context is not available directly
+            if (string.IsNullOrEmpty(startContext))
+            {
+                _logger.LogInformation("AssignmentContext not available in ContainerStartContext");
+                // Check if the context is available in blob
+                var sasUri = _settingsManager.GetSetting(EnvironmentSettingNames.ContainerStartContextSasUri);
+
+                if (!string.IsNullOrEmpty(sasUri))
+                {
+                    _logger.LogInformation("AssignmentContext ContainerStartContextSasUri available");
+                    startContext = await GetAssignmentContextFromSasUri(sasUri);
+                }
+            }
+
+            if (!string.IsNullOrEmpty(startContext))
+            {
+                _logger.LogInformation("Assigning HostAssignmentContext.");
+
+                var encryptedAssignmentContext = JsonConvert.DeserializeObject<EncryptedHostAssignmentContext>(startContext);
+                var containerKey = _settingsManager.GetSetting(EnvironmentSettingNames.ContainerEncryptionKey);
+                var assignmentContext = encryptedAssignmentContext.Decrypt(containerKey);
+                if (_instanceManager.StartAssignment(assignmentContext))
+                {
+                    _logger.LogInformation("Assign HostAssignmentContext success");
+                }
+                else
+                {
+                    _logger.LogError("Assign HostAssignmentContext failed");
+                }
+            }
+            else
+            {
+                _logger.LogInformation("Waiting for /assign to receive AssignmentContext");
```

Comment:
nit: "No host context specified. Waiting for host assignment"

Reply:
Fixed